### PR TITLE
Use better naming: status + header

### PR DESCRIPTION
**Pull request by wilzbach**

### CODE REVIEW

File: app/handlers/Exec.py [hunk 1]
```diff
@@ -258,7 +258,8 @@ def _callback(self, chunk):
                                 'application/json; charset=utf-8')
                 self.write(json.dumps(ins['data']['content']))
                 self.flush()
-            elif command == 'set_status':
+            # 'set_status' is deprecated
+            elif command == 'status' or command == 'set_status':
```
Comment (wilzbach) [inline app/handlers/Exec.py]: There's only one command for status.

File: app/handlers/Exec.py [hunk 2]
```diff
@@ -258,7 +258,8 @@ def _callback(self, chunk):
...
                 self.set_status(ins['data']['code'])
             elif command == 'set_cookie':
                 # name, value, domain, expires, path, expires_days, secure
```
Comment (wilzbach) [inline app/handlers/Exec.py]: The cookie methods are currently not exposed. Any reason?

File: app/handlers/Exec.py [hunk 3]
```diff
@@ -272,8 +273,14 @@ def _callback(self, chunk):
             elif command == 'clear_all_cookie':
                 # domain, path
                 self.clear_cookie(**ins['data'])
-            elif command == 'set_header':
-                self.set_header(ins['data']['key'], ins['data']['value'])
+            # 'set_header' is deprecated
+            elif command == 'header' or command == 'set_header':
```
Comment (wilzbach) [inline app/handlers/Exec.py]: There used to be `get_header`, but headers can now be access via `req.header`.

File: microservice.yml [hunk 4]
```diff
@@ -176,29 +176,24 @@ actions:
                   type: map
                   in: responseBody
             flush:
+              help: Flushes the current output buffer to the network.
```
Comment (wilzbach) [inline microservice.yml]: Is there even a use case for this? After all, the app can only write to Tornado via the JSON CloudEvents.

File: microservice.yml [hunk 5]
```diff
@@ -176,29 +176,24 @@ actions:
...
               http:
                 <<: *response_http
-            get_header:
```
Comment (wilzbach) [inline microservice.yml]: Headers can be accessed via `.headers`.

File: microservice.yml [hunk 6]
```diff
@@ -176,29 +176,24 @@ actions:
...
-                key:
-                  type: string
-                  in: responseBody
-                  required: true
-            set_header:
-              http:
-                <<: *response_http
-              arguments:
-                key:
+                name:
```
Comment (wilzbach) [inline microservice.yml]: Header name is the official term (no one says header key).

File: app/handlers/Exec.py (hunk 2 above)
Comment (judepereira) [reply app/handlers/Exec.py]: No idea why set_cookie is not exposed.

File: microservice.yml (hunk 4 above)
Comment (judepereira) [reply microservice.yml]: Yes. One can write data and flush it intermittently.

File: microservice.yml (hunk 5 above)
Comment (judepereira) [reply microservice.yml]: Keep it for backwards compatibility?

File: microservice.yml (hunk 6 above)
Comment (judepereira) [reply microservice.yml]: Lol - backwards compatibility?

File: microservice.yml (hunk 5 above)
Comment (wilzbach) [reply microservice.yml]: This was never supported in the first place. There's no implementation for it.

File: microservice.yml (hunk 6 above)
Comment (wilzbach) [reply microservice.yml]: This was never working / implemented.

File: microservice.yml (hunk 5 above)
Comment (judepereira) [reply microservice.yml]: Ah okay.

**Comment by wilzbach** [comment]
> @wilzbach This looks good, but I have just one concern - does `request header name: 'x-foo' value: 'a'` read better than `request setHeader name: 'x-foo' value: 'a'`? Since I do understand that `setHeader` is a little misleading, perhaps `addHeader`?

Fair enough. I renamed it.

**Comment by judepereira** [comment]
@wilzbach Can I also extend my argument for `status`? Let it be `setStatus`?

**Comment by wilzbach** [comment]
> @wilzbach Can I also extend my argument for status? Let it be setStatus?

I personally prefer `req status`, but I changed it anyhow. Note that the CI is failing as I recently added CircleCi (https://github.com/storyscript/http/pull/67 - not in master) and this PR doesn't have a config yet.

**Comment by judepereira** [comment]
Got it - merge away then.

**Pull request merged**